oeps/oep-0044-bp-changelogs.rst

```diff
@@ -0,0 +1,212 @@
+
+    * Optionally use sub-bullet for additional text or links about this change.
+
+  * **BREAKING CHANGE**: Remove deprecated feature YYY.
+
+
+  2020-03-14
+  ----------
+
+  * Added feature x. See announcement here: https://xxx
+  * Deprecated function XXX. See DEPR-XXX for details.
+
+For a repository with versioned code, use the version in the headers like::
+
+  [1.0.0] - 2020-03-21
+  --------------------
+
+* You may include an optional section at the top to collect changes that aren't yet released, although we typically release every change
+
+::
+
+  [Unreleased]
+  ------------
+
+Commit Message vs Changelog Entry
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Since changelog messages are for developers and consumers of your code, good changelog messages will often not match commit messages. Here are some example updated messages.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 50 50
+
+   * - Commit Message
+     - Changelog Entry
+   * - deps: update dependency css-loader to v3.5.1
+     - No functional change
+
+       Note: This example changelog entry assumes there were no other changes for this version.
+   * - Fix SyntaxError in UserLogout class
+     - Fix 500 error during logout
+   * - **BREAKING CHANGE** Remove the dry_run parameter in the public foobarize API method. This parameter is deprecated in favour of the no_apply parameter. See docs for details.
+     - Rename dry_run parameter
+   * - Add a set_foobarizer method to Foo's public API. This is particularly useful for developers trying to foobarize their users. See docs for details.
+     - Add set_foobarizer method to api.Foo
+
+Github Releases
+~~~~~~~~~~~~~~~
+
+* Github Releases should not replace the CHANGELOG.rst.
+* If used, Github Releases should stay in sync with the CHANGELOG.rst.
+
+  * It may be simplest to avoid them altogether, or
+  * You could publish each release with a link to the relevant version in the CHANGELOG.rst.
+
+Tooling and Automation
+======================
+
+* Use a pull-request template to remind people of any manual steps required.
+* Many Open edX frontend libraries are using `semantic-release`_ to automate the release process.
+
+  * Uses specially formatted commit messages. This may provide a good start, but see `Commit Message vs Changelog Entry`_ for recommendations around language.
+  * According to docs, `semantic-release`_ could be configured to produce Changelogs over Github Releases.
+
+* Other automation requires further research and can later be included here.
+
+.. _semantic-release: https://github.com/semantic-release/semantic-release
+
+Consequences
+============
+
+Since these are only recommendations, strictly speaking, there are no consequences.
+
+If you want to follow the best practice, update your repository to follow this guidance.
+
+Rejected Alternatives
+=====================
+
+For repositories without versioned code, like edx-platform and other IDAs, the only real alternatives to a CHANGELOG.rst are:
```

Comment:
What would it take to add a CHANGELOG.rst to edx-platform? I understand that we can't count on version numbers, and release names are too far apart to simply dump changes in a single file. But I believe @nedbat had in mind some tooling to facilitate the addition of changelog entries. Maybe these changes could be aggregated every now and then (every week? month?) in a single changelog file for other users to keep track of?